src/test/java/uk/gov/hmcts/divorce/citizen/notification/ApplicationSentForReviewApplicant2NotificationTest.java

```diff
@@ -0,0 +1,121 @@
+
+    @Mock
+    private EmailTemplatesConfig emailTemplatesConfig;
+
+    @Mock
+    private CommonContent commonContent;
+
+    @InjectMocks
+    private ApplicationSentForReviewApplicant2Notification notification;
+
+    @Test
+    void shouldSendEmailToApplicant2WhileInAwaitingApplicant2ResponseState() {
+        CaseData data = caseData();
+        data.setApplicant2(getApplicant2(Gender.MALE));
+        data.setDueDate(LOCAL_DATE);
+        final HashMap<String, String> templateVars = new HashMap<>();
+
+        when(commonContent.templateVarsFor(data)).thenReturn(templateVars);
+        when(commonContent.getTheirPartner(data, data.getApplicant1())).thenReturn("husband");
+        when(emailTemplatesConfig.getTemplateVars()).thenReturn(getConfigTemplateVars());
+
+        notification.send(data, 1234567890123456L);
+
+        verify(notificationService).sendEmail(
+            eq(TEST_USER_EMAIL),
+            eq(JOINT_APPLICANT2_ANSWERS_SENT_FOR_REVIEW),
+            argThat(allOf(
+                hasEntry(SUBMISSION_RESPONSE_DATE, LOCAL_DATE.format(dateTimeFormatter)),
+                hasEntry(PARTNER, "husband"),
+                hasEntry(CREATE_ACCOUNT_LINK, SIGN_IN_DIVORCE_TEST_URL)
+            )),
+            eq(ENGLISH)
+        );
+        verify(commonContent).templateVarsFor(data);
+        verify(commonContent).getTheirPartner(data, data.getApplicant1());
+    }
+
+    @Test
+    void shouldSetTheAppropriateFieldsForDissolutionCases() {
+        CaseData data = caseData();
+        data.setApplicant2(getApplicant2(Gender.MALE));
+        data.setDueDate(LOCAL_DATE);
+        data.setDivorceOrDissolution(DivorceOrDissolution.DISSOLUTION);
+        final HashMap<String, String> templateVars = new HashMap<>();
+
+        when(commonContent.templateVarsFor(data)).thenReturn(templateVars);
+        when(commonContent.getTheirPartner(data, data.getApplicant1())).thenReturn("husband");
+        when(emailTemplatesConfig.getTemplateVars()).thenReturn(getConfigTemplateVars());
+
+        notification.send(data, 1234567890123456L);
+
+        verify(notificationService).sendEmail(
+            eq(TEST_USER_EMAIL),
+            eq(JOINT_APPLICANT2_ANSWERS_SENT_FOR_REVIEW),
+            argThat(allOf(
+                hasEntry(SUBMISSION_RESPONSE_DATE, LOCAL_DATE.format(dateTimeFormatter)),
+                hasEntry(PARTNER, "husband"),
+                hasEntry(CREATE_ACCOUNT_LINK, SIGN_IN_DISSOLUTION_TEST_URL)
+            )),
+            eq(ENGLISH)
+        );
+        verify(commonContent).templateVarsFor(data);
+        verify(commonContent).getTheirPartner(data, data.getApplicant1());
+    }
+
+    @Test
+    void shouldNotSendEmailForInvalidCaseData() {
+        CaseData data = caseData();
+        final HashMap<String, String> templateVars = new HashMap<>();
+
+        when(commonContent.templateVarsFor(data)).thenReturn(templateVars);
+
+        assertThrows(NullPointerException.class, () ->
```

Comment:
getting null pointer exception is not good instead why not handle this scenario in the send method

Reply:
@nitinprabhu would it be better to set all the template variables to a blank string first? Or could use a try/catch around the notificationService.sendEmail()

Reply:
if you expect the template vars to be set then it should be present else an error should be thrown instead of setting it to blank.
may be try/catch should be fine for now. Before sending email I assume we validate the case data if not something we can discuss with Linus and see if we want to add it

Reply:
I've added in some basic case data validation